test/test_image.py

```diff
@@ -258,6 +264,41 @@ def test_write_file_non_ascii():
         assert content == saved_content
 
 
+@pytest.mark.skipif(cv2 is None, reason='opencv unavailable')
+@pytest.mark.parametrize('shape', [
+    (27, 27),
+    (60, 60),
+    (105, 105),
+])
+def test_read_1_bit_png(shape):
+    with get_tmp_dir() as root:
+        image_path = os.path.join(root, 'test_{shape}.png')
+        img = Image.new('1', shape)
```

Comment:
This generates an image full of zeros, which wouldn't show the original issue that this PR is fixing (namely, we were reading 8 1bit pixels and "casting" it to a single 8bit pixel.

Can you instead do something like
```python
img = np.random.rand(10, 10) > 0.5
img = Image.fromarray(img)
img.save(image_path)

img1 = read_image(image_path)
torch.testing.assert_close(img1, torch.as_tensor(img), check_stride=False)
```
so that we can:
- avoid the dependency on OpenCV
- have a more robust test as the data is not only zeros?

Reply:
Makes sense, thanks!